src/renderer/components/+namespaces/namespace-select.tsx

```diff
@@ -82,6 +82,22 @@ export class NamespaceSelect extends React.Component<Props> {
     );
   };
 
+  filterOption = (option: SelectOption, rawInput: string) => {
+    if (option.value === "" || (!namespaceStore.areAllSelectedImplicitly && namespaceStore.selectedNames.has(option.value))) {
+      return true;
+    }
+
+    if (namespaceStore.items.length > 500 && rawInput.length < 3) {
```

Comment:
_nut:_ Maybe better to extract `namespaceStore.items.length > 500 && rawInput.length < 3` to some variable with meaning name? Would add better understanding of why it quits the function. (+similar for line:94)